tools/depcheck/pkg/graph/flags.go

```diff
@@ -0,0 +1,317 @@
+				glog.V(1).Infof("Skipping unvisited (missing) dependency %q, which is imported by package %q", dependency, pkg.ImportPath)
+				continue
+			}
+
+			if g.HasEdgeFromTo(from, to) {
+				continue
+			}
+
+			g.SetEdge(concrete.Edge{
+				F: from,
+				T: to,
+			}, 0)
+		}
+	}
+
+	// filter graph if any filters are specified
+	if len(o.Filters) > 0 {
+		var err error
+		g, err = FilterPackages(g, o.Filters)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	g.PruneOrphans()
+	return g, nil
+}
+
+type GraphFlags struct {
+	Filter  string
+	Exclude string
+
+	Openshift bool
+
+	RepoImportPath string
+	Entrypoints    []string
+}
+
+// calculate roots receives a set of entrypoints and traverses through
+// the directory tree, returning a list of all reachable go packages.
+// Excludes the vendor tree.
+// Returns the list of calculated import paths or an error.
+func (o *GraphFlags) calculateRoots() ([]string, error) {
+	packages, err := GetPackageMetadata(
+		ensureEntrypointEllipsis(o.Entrypoints),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	roots := []string{}
+	for _, pkg := range packages.Packages {
+		roots = append(roots, pkg.ImportPath)
+	}
+
+	return roots, nil
+}
+
+func (o *GraphFlags) ToOptions(out, errout io.Writer) (*GraphOptions, error) {
+	opts := &GraphOptions{
+		RepoImportPath: o.RepoImportPath,
+	}
+
+	if len(o.Entrypoints) == 0 {
+		return nil, errors.New("at least one entrypoint path must be provided")
+	}
+	if o.Openshift && (len(o.Exclude) > 0 || len(o.Filter) > 0) {
+		return nil, errors.New("--exclude or --filter are mutually exclusive with --openshift")
+	}
+
+	// calculate go package info from given set of entrypoints
+	packages, err := GetPackageMetadata(
+		ensureEntrypointEllipsis(
+			ensureVendorEntrypoint(o.Entrypoints, o.RepoImportPath),
+		),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	opts.Packages = packages
+
+	// calculate non-vendor trees
+	roots, err := o.calculateRoots()
+	if err != nil {
+		return nil, err
+	}
+	opts.Roots = roots
+
+	// set openshift defaults
+	if o.Openshift {
+		opts.Excludes = getOpenShiftExcludes()
+		filters, err := getOpenShiftFilters()
+		if err != nil {
+			return nil, err
+		}
+
+		opts.Filters = filters
+		return opts, nil
+	}
+
+	if len(o.Exclude) > 0 {
+		f, err := ioutil.ReadFile(o.Exclude)
+		if err != nil {
+			return nil, err
+		}
+
+		err = json.Unmarshal(f, &opts.Excludes)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if len(o.Filter) > 0 {
+		f, err := ioutil.ReadFile(o.Filter)
+		if err != nil {
+			return nil, err
+		}
+
+		err = json.Unmarshal(f, &opts.Filters)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return opts, nil
+}
+
+func (o *GraphFlags) AddFlags(cmd *cobra.Command) {
+	cmd.Flags().BoolVar(&o.Openshift, "openshift", o.Openshift, "generate and use OpenShift-specific lists of excluded packages and filters.")
+	cmd.Flags().StringVar(&o.RepoImportPath, "root", o.RepoImportPath, "filepath of root repository to analyze")
```

Comment:
usability-wise, seems like you should force them use `...` if they really want it.  You're constraining choice here.

Reply:
Would this be for cases where a user wants to begin traversing from the base of the repo (instead of a subpackage)?